torchtext/_download_hooks.py

```diff
@@ -1,15 +1,20 @@
 import re
+from functools import partial
 
 import requests
 
 # This is to allow monkey-patching in fbcode
-from torch.hub import load_state_dict_from_url  # noqa
+from torch.hub import load_state_dict_from_url as torchhub_load_state_dict_from_url  # noqa
+from torchtext import _CACHE_DIR
 from torchtext._internal.module_utils import is_module_available
 from tqdm import tqdm
 
 if is_module_available("torchdata"):
     from torchdata.datapipes.iter import HttpReader, GDriveReader  # noqa F401
 
+# we set the model_dir to be equal to _CACHE_DIR so that all checkpoint downloads go into the text specific cache
+load_state_dict_from_url = partial(torchhub_load_state_dict_from_url, model_dir=_CACHE_DIR)
```

Comment:
The use of partial makes it looks like a hack to me. 
But IIUC, this approach is inlined with decisions on other so it seems better to be a regular function.

Reply:
On a separate note, `torchtext._CACHE_DIR` is a constant string, so this won't allow client code to change the cache directory at runtime. That seems to be inflexible to me. Perhaps get rid of `torchtext._CACHE_DIR` and change it to a function.

Reply:
Consolidating directories for caches is good, however I'd keep pre-trained models and dataset caches in separate sub directory.

Reply:
I ended up deciding to scrap this change alltogether and just follow the approach torchaudio takes. This means that all model checkpoints will still be downloaded to the `torch.hub.home` directory.

I think it should be fine for `torchtext._CACHE_DIR` to be a constant string as it's only used to download dataset and other asset files from a given URL (not model checkpoints). If clients wanted to modify the root part of `torchtext._CACHE_DIR`, they can just update the `$TORCH_HOME` env variable as that is respected by the `_get_torch_home()` method.